content/shmem_team_split_3d.tex

```diff
@@ -39,24 +43,49 @@
 \end{apiarguments}
 
 \apidescription{
-The shmemx\_team\_split\_3d routine is a collective routine. It
-partitions an existing parent team into three subgroups, based on the
-three-dimensional Cartesian space defined by the triplet (xrange,
-yrange, and zrange) describing the size of the Cartesian space in X,
-Y, and Z dimensions. Each subgroup contains all \acp{PE} that are in the same
-dimension, along the X-axis, Y-axis and Z-axis. Within each subgroup,
-the \acp{PE} are ranked based on the position of the \ac{PE} with respect to its
+The shmem\_team\_split\_3d routine is a collective routine. It creates up to three new teams
+by splitting an existing parent team into up to three subsets based on a
+1D, 2D or 3D Cartesian space defined by the triplet (xrange, yrange, and zrange)
```

Comment:
The reference to a 1D Cartesian space made me wonder: Do we permit the case where `yrange` and `zrange` are both zero? If so, what is the resulting team?

Reply:
Also: `xrange` &#8594; `\VAR{xrange}` (and similarly for `yrange` and `zrange`)

Reply:
Ah, the 1D reference was because I considered explicitly describing the case where we have xrange=N, yrange=0 and zrange=0, but this should be the same result as split_strided where start = 0, stride = 1, size = N, with the xaxis_team being the same as the new_team from a strided split with those parameters, right?

And if yrange is allowed to be 0, can we legally do an NxM 2D split where xrange=N, yrange=0, zrange=M? Or if yrange=0 does zrange also have to be 0?

Maybe it is better to only allow 2D and 3D split here if the 1D split would just be a strided split with stride of 1.